src/components/AirbaseNG/AirbaseNG.tsx

```diff
@@ -1,165 +1,171 @@
+    const [opened, setOpened] = useState(!isCommandAvailable);
+    const [loadingModal, setLoadingModal] = useState(true);
     const [advanceMode, setAdvanceMode] = useState(false);
     const [customMode, setCustomMode] = useState(false);
 
-    // Component Constants.
-    const title = "Airbase-ng"; // Title of the component.
-    const description = "Airbase-ng is a tool to create fake access points."; // Description of the component.
+    // Constants for the component's metadata.
+    const title = "Airbase-ng";
+    const description = "Airbase-ng is a tool to create fake access points.";
     const steps =
         "Step 1: Type in the name of your fake host.\n" +
         "Step 2: Select your desired channel.\n" +
         "Step 3: Specify the WLAN interface to be used.\n" +
         "Step 4: Click 'Start AP' to begin the process.\n" +
-        "Step 5: View the output block to see the results. ";
-    const sourceLink = ""; // Link to the source code (or Kali Tools).
-    const tutorial = ""; // Link to the official documentation/tutorial.
-    const dependencies = ["aircrack-ng"]; // Contains the dependencies required by the component.
+        "Step 5: View the output block to see the results.";
+    const sourceLink = ""; // Placeholder for source link.
+    const tutorial = ""; // Placeholder for tutorial link.
+    const dependencies = ["aircrack-ng"];
 
-    // Form hook to handle form input.
+    // Form handling for the component's input fields.
     const form = useForm({
         initialValues: {
             fakeHost: "",
-            channel: "",
-            replayInterface: "",
+            channel: "6", // Default channel
+            replayInterface: "wlan0", // Default interface
             macAddress: "",
             filePath: "",
+            verbosity: false,
+            disableBeacons: false,
             customConfig: "",
+            customScript: "",
+        },
+
+        // Validation: Ensure essential fields are filled.
+        validate: {
+            fakeHost: (value) => (value ? null : "Fake host name is required."),
+            channel: (value) => (value ? null : "Channel is required."),
+            replayInterface: (value) => (value ? null : "WLAN interface is required."),
         },
     });
 
-    // Check if the command is available and set the state variables accordingly.
+    // Effect to check command availability on component mount.
     useEffect(() => {
-        // Check if the command is available and set the state variables accordingly.
         checkAllCommandsAvailability(dependencies)
             .then((isAvailable) => {
-                setIsCommandAvailable(isAvailable); // Set the command availability state
-                setOpened(!isAvailable); // Set the modal state to opened if the command is not available
-                setLoadingModal(false); // Set loading to false after the check is done
+                setIsCommandAvailable(isAvailable);
```

Comment:
Should keep the comment. 
```suggestion
                setIsCommandAvailable(isAvailable); // Set the command availability state
```